packages/recurring-payments-job/src/__tests__/recurring-payments-processor.spec.js

```diff
@@ -68,10 +82,149 @@ describe('recurring-payments-processor', () => {
+
+      expect(errorSpy).toHaveBeenCalledWith('Run aborted. Error fetching due recurring payments:', error)
+    })
+  })
+
+  describe('When payment request throws an error...', () => {
+    it('debug is called with error message', async () => {
+      salesApi.getDueRecurringPayments.mockReturnValueOnce(getMockPaymentRequestResponse())
+      const oopsie = new Error('payment gate down')
+      sendPayment.mockRejectedValueOnce(oopsie)
+
+      try {
+        await processRecurringPayments()
+      } catch {}
+
+      expect(debugLogger).toHaveBeenCalledWith(expect.any(String), oopsie)
+    })
+
+    it('prepares and sends all payment requests, even if some fail', async () => {
+      // arrange...
+      const agreementIds = [Symbol('agreementId1'), Symbol('agreementId2'), Symbol('agreementId3'), Symbol('agreementId4')]
+      salesApi.getDueRecurringPayments.mockReturnValueOnce([
+        getMockDueRecurringPayment('fee', agreementIds[0]),
+        getMockDueRecurringPayment('fi', agreementIds[1]),
+        getMockDueRecurringPayment('foe', agreementIds[2]),
+        getMockDueRecurringPayment('fum', agreementIds[3])
+      ])
+
+      const permissionData = { licensee: { countryCode: 'GB-ENG' } }
+      for (let x = 0; x < agreementIds.length; x++) {
+        salesApi.preparePermissionDataForRenewal.mockReturnValueOnce(permissionData)
+        salesApi.createTransaction.mockReturnValueOnce({
+          cost: 50,
+          id: `transaction-id-${x + 1}`
+        })
+
+        if (x === 1) {
+          const err = new Error('Payment request failed')
+          sendPayment.mockRejectedValueOnce(err)
+        } else {
+          sendPayment.mockResolvedValueOnce({ payment_id: `test-payment-id-${x + 1}`, agreementId: agreementIds[x] })
+        }
+        if (x < 3) {
+          getPaymentStatus.mockResolvedValueOnce(getPaymentStatusSuccess())
+        }
+      }
+      const expectedData = {
+        amount: 5000,
+        description: 'The recurring card payment for your rod fishing licence',
+        reference: 'transactionId',
+        authorisation_mode: 'agreement'
+      }
+
+      // act...
+      await processRecurringPayments()
+
+      // assert...
+      expect(sendPayment).toHaveBeenCalledTimes(4)
+      expect(sendPayment).toHaveBeenNthCalledWith(
+        1,
+        expect.objectContaining({ ...expectedData, reference: 'transaction-id-1', agreement_id: agreementIds[0] })
+      )
+      expect(sendPayment).toHaveBeenNthCalledWith(
+        2,
+        expect.objectContaining({ ...expectedData, reference: 'transaction-id-2', agreement_id: agreementIds[1] })
+      )
+      expect(sendPayment).toHaveBeenNthCalledWith(
+        3,
+        expect.objectContaining({ ...expectedData, reference: 'transaction-id-3', agreement_id: agreementIds[2] })
+      )
+      expect(sendPayment).toHaveBeenNthCalledWith(
+        4,
+        expect.objectContaining({ ...expectedData, reference: 'transaction-id-4', agreement_id: agreementIds[3] })
```

Comment:
This test is quite complex and has a lot of different assertions. Would it be worth splitting it out into multiple tests and refactoring some of the setup into a different function?